kinoml/features/protein_struct_features.py

```diff
@@ -0,0 +1,185 @@
+"""
+protein.py
+This is a tool to featurize kinase conformational changes through the entire Kinome.
+
+"""
+
+def key_klifs_residues(numbering):
+    """
+    Retrieve a list of PDB residue indices relevant to key kinase conformations mapped via KLIFS.
+
+    Define indices of the residues relevant to a list of 12 collective variables relevant to
+    kinase conformational changes. These variables include: angle between aC and aE helices,
+    the key K-E salt bridge, DFG-Phe conformation (two distances), X-DFG-Phi, X-DFG-Psi,
+    DFG-Asp-Phi, DFG-Asp-Psi, DFG-Phe-Phi, DFG-Phe-Psi, DFG-Phe-Chi1, and the FRET L-S distance.
+    All features are under the current numbering of the structure provided.
+
+    Parameters
+    ----------
+    numbering : list of int
+        numbering[klifs_index] is the residue number for the given PDB file corresponding to KLIFS residue index 'klifs_index'
+
+    Returns
+    -------
+    key_res : list of int
+        Key residue indices
+
+    """
+    if numbering == None:
+        print("The structure was not found in the klifs database.")
+        key_res = None
+        return key_res
+
+    key_res = dict() #initialize key_res (which read from the 0-based numbering list)
+    for i in range(5):
+        key_res[f'group{i}'] = list()
+    ## feature group 0: A-loop backbone dihedrals
+    key_res['group0'].append(numbering[83]) # start of A-loop
+
+    ## feature group 1: P-loop backbone dihedrals
+    key_res['group1'].append(numbering[3]) # res0 in P-loop
```

Comment:
These magic indices are bit concerning if they are not well documented. I guess they are residue indices?